### ISSUE TITLE
sync problem with file with lower case and upper case

### ISSUE THREAD

**Issue by StanLepunK** [isaacs/github#1520]
Hello,
the name file is not changed when it's just type case is changed.
That can cause a trouble when we come back to old version or when other people download the file.
Or maybe I miss something to update the filename when the changement is only a upper case or lowr case.
<img width="195" alt="Capture d’écran 2019-03-13 à 11 43 57" src="https://user-images.githubusercontent.com/3332269/54273849-ffbcac80-4586-11e9-9b70-d9d1281f3e36.png">
<img width="204" alt="Capture d’écran 2019-03-13 à 11 45 58" src="https://user-images.githubusercontent.com/3332269/54273855-04816080-4587-11e9-8184-f92003e16489.png">

**Comment by StanLepunK** [comment]
Maybe more explicit:
<img width="781" alt="Capture d’écran 2019-03-13 à 12 26 14" src="https://user-images.githubusercontent.com/3332269/54275603-5926da80-458b-11e9-85b6-fca5531fd210.png">

**Comment by oprogramador** [comment]
@StanLepunK 

It's not about GitHub. It a local problem on Mac.

A workaround if you want to change the case
- remove all the files which case you want to change
- `git add .`
- `git commit -m 'remove files'`
- for each file `git checkout your-branch your-file`
- for each file `mv UPPERCASE_NAME lowercase_name`
- `git add .`
- `git commit -m 'add files'`
- `git rebase -i HEAD~2`
- for the second commit, replace `pick` with `squash`
- write down your desired commit message

**Comment by StanLepunK** [comment]
hmmm thx for the advice but i'm not really confortable with this type of operation: I figure, I must do that in the terminal to targeting the folder of my project ?

**Comment by TPS** [comment]
@StanLepunK Did you contact GitHub support?  What'd they say?

**Comment by StanLepunK** [comment]
Yep I contact us, but I don't have a time now to solve it, but i do that asap. If I understand well the support say the solution of @oprogramador is not really optimum because that can erase my file history. the fix proposition is : 


Hey there Stan,
Thanks for writing in and I hope you are having a great day where you are.Taking a look at that issue, as the user mentions, it looks like it might be a local issue with your repository.
Can you try, locally, running:`$ git mv :old_filename :new_filename` Then run `$ git commit -a -m ":commit_message"` By using the `git mv` command you are telling git to move/rename the file - and by doing so continue to track the changes. In https://github.com/isaacs/github/issues/1520  the commenter has mentioned removing and readding the files to git,  this will destroy the history of these files - which is still a valid  method if that's the desired process you have.
I hope this helps and please let me know if you face any further issues with this.Thanks Matt Barber

**Comment by TPS** [comment]
That seems very odd that @GitHub would treat it like that.…  But I suppose they're making the easy assumption.  Still, [the appropriate section @ Wikipedia](https://en.wikipedia.org/wiki/Case_sensitivity#In_filesystems) currently says _(emphasis mine)_
> In filesystems in Unix-like systems, filenames are usually case-sensitive (there can be separate readme.txt and Readme.txt files in the same directory). MacOS is somewhat unusual in that, by default, it uses HFS+ in a case-insensitive (so that there cannot be a readme.txt and a Readme.txt in the same directory) but case-preserving mode (so that a file created as readme.txt is shown as readme.txt and a file created as Readme.txt is shown as Readme.txt) by default. ___This causes some issues for developers and power users,___ because most other environments are case-sensitive, but many Mac Installers fail on case-sensitive file systems.

**Comment by oprogramador** [comment]
@StanLepunK @TPS 

Anyway I think, it's not a GitHub issue. It's just about how git and operating systems work.

**Comment by TPS** [comment]
@oprogramador I disagree.  The git|GitHub client should preserve case, even in cases when the OS &/or filesystem _don't_.  Since MacOS _does_, I don't understand the above behavior @ all.

**Comment by user4i** [comment]
📌📌📌 
If you need to commit files renaming, just capitalization, and you have a problems with staging/commiting, as option

1. Change capitalization and add "1" to the end (or other symbol)
2. Stage (without commit) = result "R" - renaming that files
3. Remove "1" from the end 
4. Stage (without commit) = result "R" - renaming that files (only capitalization now)
5. Do commit.

**Comment by StanLepunK** [comment]
thanks for the tip

**Comment by IVNtech** [comment]
> 📌📌📌
> If you need to commit files renaming, just capitalization, and you have a problems with staging/commiting, as option
> 
> 1. Change capitalization and add "1" to the end (or other symbol)
> 2. Stage (without commit) = result "R" - renaming that files
> 3. Remove "1" from the end
> 4. Stage (without commit) = result "R" - renaming that files (only capitalization now)
> 5. Do commit.

THANKS THIS DO THE JOB DONE

**Comment by coolzoom** [comment]
> 📌📌📌
> If you need to commit files renaming, just capitalization, and you have a problems with staging/commiting, as option
> 
> 1. Change capitalization and add "1" to the end (or other symbol)
> 2. Stage (without commit) = result "R" - renaming that files
> 3. Remove "1" from the end
> 4. Stage (without commit) = result "R" - renaming that files (only capitalization now)
> 5. Do commit.

Epic tips! Thanks. 
also, I found no need stage if you use git mv to rename.
for anyone who wish to try:

Option 1 example based on @user4i :
> mv Folder folder1
> git stage .
> mv folder1 folder
> git stage .
> git add .
> git commit -a -m ":commit_message"

Option 2 example based on https://github.com/isaacs/github/issues/1520#issuecomment-473229740
> git mv -f Folder folder1
> git mv -f folder1 folder
> git add .
> git commit -a -m ":commit_message"

**Comment by user4i** [comment]
> also, I found no need stage if you use git mv to rename.
> for anyone who wishes to try, below are the examples of renaming a folder.

Good idea! But stage is more convenient and quick to apply this steps https://github.com/isaacs/github/issues/1520#issuecomment-570152685 in visual code editor like vscode or visual git tool. Additionally Option 2 was not so easy to apply, may be because of my OS/git settings.
